src/components/user-group/repo/classes.repo.integration.test.js

```diff
@@ -0,0 +1,129 @@
+		});
+
+		it('should return empty array for user with no class assigned', async () => {
+			const student = await testObjects.createTestUser({ roles: ['student'] });
+			const classes = await classesRepo.getClassesForUser(student._id);
+			expect(classes.length).to.be.equal(0);
+		});
+	});
+
+	describe('when find classes for teacher', async () => {
+		it('should return list of classes where the teacher is part of ', async () => {
+			const { _id: schoolId } = await testObjects.createTestSchool();
+			const teacher = await testObjects.createTestUser({ roles: ['teacher'], schoolId });
+			const testClass = await testObjects.createTestClass({ teacherIds: [teacher._id], schoolId });
+
+			const classes = await classesRepo.findClassesByTeacher(teacher._id);
+
+			expect(classes[0]._id).to.deep.equal(testClass._id);
+			expect(classes[0].teacherIds[0]).to.deep.equal(teacher._id);
+		});
+
+		it('should return empty array for user with no class assigned (as teacher)', async () => {
+			const teacher = await testObjects.createTestUser({ roles: ['teacher'] });
+			const classes = await classesRepo.findClassesByTeacher(teacher._id);
+			expect(classes.length).to.be.equal(0);
+		});
+	});
+
+	describe('when find classes for student', async () => {
+		it('should return list of classes where the student is part of ', async () => {
+			const { _id: schoolId } = await testObjects.createTestSchool();
+			const student = await testObjects.createTestUser({ roles: ['student'], schoolId });
+			const testClass = await testObjects.createTestClass({ userIds: [student._id], schoolId });
+
+			const classes = await classesRepo.findClassesByStudent(student._id);
+
+			expect(classes[0]._id).to.deep.equal(testClass._id);
+			expect(classes[0].userIds[0]).to.deep.equal(student._id);
+		});
+
+		it('should return empty array for user with no class assigned (as student)', async () => {
+			const student = await testObjects.createTestUser({ roles: ['student'] });
+			const classes = await classesRepo.findClassesByStudent(student._id);
+			expect(classes.length).to.be.equal(0);
+		});
+	});
+
+	describe('delete user from classes', () => {
+		it('should return list of updated classes', async () => {
+			const { _id: schoolId } = await testObjects.createTestSchool();
+			const student = await testObjects.createTestUser({ roles: ['student'], schoolId });
+			await testObjects.createTestClass({ userIds: [student._id], schoolId });
+
+			const result = await classesRepo.removeUserFromClasses(student._id);
+
+			expect(result.n).to.be.equal(1);
+			expect(result.nModified).to.be.equal(1);
+		});
+
+		it('the class should not contain the student after delete', async () => {
+			const { _id: schoolId } = await testObjects.createTestSchool();
+			const student = await testObjects.createTestUser({ roles: ['student'], schoolId });
+			const testClass = await testObjects.createTestClass({ userIds: [student._id], schoolId });
+
```

Comment:
here i would expect
```
const result = await classesRepo.findClassById(testClass._id);
expect(result.userIds.map(idToString)).to.include(idToString(student._id));
```

Reply:
filterClassMember is used for the find and deletion task. If use repo methods on repo tests we can changes this query helper and the test always true.